stdlib/public/core/ArrayCast.swift

```diff
@@ -152,7 +156,7 @@ internal func _arrayConditionalBridgeElements<SourceElement, TargetElement>(
   let buf = _ContiguousArrayBuffer<TargetElement>(
     uninitializedCount: source.count, minimumCapacity: 0)
   
-  var p = buf.firstElementAddress
+  var p = buf.firstElementAddress!
```

Comment:
I honestly wasn't sure when the distinction was important to the surrounding code. Do you want me to go back and reanalyze which ones make sense as `firstElementAddress` and which ones make sense as `subscriptBaseAddress`? If the rule is just "we're going to load from it unconditionally", then sure.